donjuan/grid.py

```diff
@@ -69,6 +69,58 @@ def __init__(
     ):
         super().__init__(n_rows, n_cols, cells)
 
+    def find_walls(self) -> List[Tuple[Tuple[int, int], Tuple[int, int], str]]:
+        """
+        Find all of the continuous walls.
+        :return: a list of walls, where a wall is defined as a three part tuple:
+                        1. the first cell of the wall's coordinates
+                        2. the last cell of the wall's coordinates
+                        3. the dimension of the wall (vertical or horizontal)
+                 vertical walls run along side the east side of the cells
+                 horizontal walls run along side the south side of the cells
+        """
+        checked_borders = {}
+        walls = []
+        for row, y in zip(self.cells, range(len(self.cells))):
+            for cell, x in zip(row, range(len(row))):
+                if (x, y, "y") not in checked_borders:
+                    checked_borders[(x, y, "y")] = 1
+                    if (
+                        y + 1 < len(self.cells)
+                        and cell.filled != self.cells[y + 1][x].filled
+                    ):
+                        wall_start = (x, y)
+                        wall_end = (x, y)
+                        i = 1
+                        while x + i < len(row):
```

Comment:
Delete the line above.

```suggestion
                        for i in range(x + 1, self.n_cols):
```